### ISSUE TITLE
VST3 linux 32 build fails

### ISSUE THREAD

**Issue by trebmuh** [ryukau/VSTPlugins#9]
Building from git master fbeb936 on a Debian stable (Buster) system in a 32bits chroot (using pbuilder), it fails with:
```
[ 59%] Building CXX object public.sdk/samples/vst-hosting/validator/CMakeFiles/validator.dir/__/__/__/source/vst/hosting/module_linux.cpp.o
In file included from /build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:27,
                 from /build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/samples/vst-hosting/validator/source/validator.cpp:42:
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::sizeofCheckProcessSetup<ProcessSetup, w, x, y, z>::sizeofCheckProcessSetup() [with ProcessSetup = Steinberg::Vst::ProcessSetup; unsigned int w = 24; unsigned int x = 20; unsigned int y = 24; unsigned int z = 24]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:49:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct sizeof error: ProcessSetup
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:44:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (sizeof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:49:1: note: in expansion of macro 'SMTG_TYPE_SIZE_CHECK'
 SMTG_TYPE_SIZE_CHECK (ProcessSetup, 24, 20, 24, 24)
 ^~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::sizeofCheckAudioBusBuffers<AudioBusBuffers, w, x, y, z>::sizeofCheckAudioBusBuffers() [with AudioBusBuffers = Steinberg::Vst::AudioBusBuffers; unsigned int w = 24; unsigned int x = 16; unsigned int y = 24; unsigned int z = 24]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:50:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct sizeof error: AudioBusBuffers
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:44:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (sizeof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:50:1: note: in expansion of macro 'SMTG_TYPE_SIZE_CHECK'
 SMTG_TYPE_SIZE_CHECK (AudioBusBuffers, 24, 16, 24, 24)
 ^~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::alignofCheckProcessSetup<ProcessSetup, w, x, y, z>::alignofCheckProcessSetup() [with ProcessSetup = Steinberg::Vst::ProcessSetup; unsigned int w = 8; unsigned int x = 1; unsigned int y = 8; unsigned int z = 8]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:52:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct alignof error: ProcessSetup
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:50:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (alignof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:52:1: note: in expansion of macro 'SMTG_TYPE_ALIGN_CHECK'
 SMTG_TYPE_ALIGN_CHECK (ProcessSetup, 8, 1, 8, 8)
 ^~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::alignofCheckAudioBusBuffers<AudioBusBuffers, w, x, y, z>::alignofCheckAudioBusBuffers() [with AudioBusBuffers = Steinberg::Vst::AudioBusBuffers; unsigned int w = 8; unsigned int x = 1; unsigned int y = 8; unsigned int z = 8]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:53:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct alignof error: AudioBusBuffers
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:50:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (alignof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:53:1: note: in expansion of macro 'SMTG_TYPE_ALIGN_CHECK'
 SMTG_TYPE_ALIGN_CHECK (AudioBusBuffers, 8, 1, 8, 8)
 ^~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::alignofCheckParameterInfo<ParameterInfo, w, x, y, z>::alignofCheckParameterInfo() [with ParameterInfo = Steinberg::Vst::ParameterInfo; unsigned int w = 8; unsigned int x = 1; unsigned int y = 8; unsigned int z = 8]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:65:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct alignof error: ParameterInfo
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:50:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (alignof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:65:1: note: in expansion of macro 'SMTG_TYPE_ALIGN_CHECK'
 SMTG_TYPE_ALIGN_CHECK (ParameterInfo, 8, 1, 8, 8)
 ^~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::sizeofCheckEvent<Event, w, x, y, z>::sizeofCheckEvent() [with Event = Steinberg::Vst::Event; unsigned int w = 48; unsigned int x = 40; unsigned int y = 40; unsigned int z = 48]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:74:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct sizeof error: Event
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:44:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (sizeof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:74:1: note: in expansion of macro 'SMTG_TYPE_SIZE_CHECK'
 SMTG_TYPE_SIZE_CHECK (Event, 48, 40, 40, 48)
 ^~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::alignofCheckEvent<Event, w, x, y, z>::alignofCheckEvent() [with Event = Steinberg::Vst::Event; unsigned int w = 8; unsigned int x = 1; unsigned int y = 8; unsigned int z = 8]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:82:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct alignof error: Event
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:50:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (alignof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:82:1: note: in expansion of macro 'SMTG_TYPE_ALIGN_CHECK'
 SMTG_TYPE_ALIGN_CHECK (Event, 8, 1, 8, 8)
 ^~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::sizeofCheckNoteExpressionValueDescription<NoteExpressionValueDescription, w, x, y, z>::sizeofCheckNoteExpressionValueDescription() [with NoteExpressionValueDescription = Steinberg::Vst::NoteExpressionValueDescription; unsigned int w = 32; unsigned int x = 28; unsigned int y = 32; unsigned int z = 32]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:84:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct sizeof error: NoteExpressionValueDescription
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:44:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (sizeof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:84:1: note: in expansion of macro 'SMTG_TYPE_SIZE_CHECK'
 SMTG_TYPE_SIZE_CHECK (NoteExpressionValueDescription, 32, 28, 32, 32)
 ^~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::sizeofCheckNoteExpressionTypeInfo<NoteExpressionTypeInfo, w, x, y, z>::sizeofCheckNoteExpressionTypeInfo() [with NoteExpressionTypeInfo = Steinberg::Vst::NoteExpressionTypeInfo; unsigned int w = 816; unsigned int x = 812; unsigned int y = 816; unsigned int z = 816]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:87:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct sizeof error: NoteExpressionTypeInfo
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:44:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (sizeof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:87:1: note: in expansion of macro 'SMTG_TYPE_SIZE_CHECK'
 SMTG_TYPE_SIZE_CHECK (NoteExpressionTypeInfo, 816, 812, 816, 816)
 ^~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::alignofCheckNoteExpressionValueDescription<NoteExpressionValueDescription, w, x, y, z>::alignofCheckNoteExpressionValueDescription() [with NoteExpressionValueDescription = Steinberg::Vst::NoteExpressionValueDescription; unsigned int w = 8; unsigned int x = 1; unsigned int y = 8; unsigned int z = 8]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:89:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct alignof error: NoteExpressionValueDescription
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:50:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (alignof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:89:1: note: in expansion of macro 'SMTG_TYPE_ALIGN_CHECK'
 SMTG_TYPE_ALIGN_CHECK (NoteExpressionValueDescription, 8, 1, 8, 8)
 ^~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::alignofCheckNoteExpressionValueEvent<NoteExpressionValueEvent, w, x, y, z>::alignofCheckNoteExpressionValueEvent() [with NoteExpressionValueEvent = Steinberg::Vst::NoteExpressionValueEvent; unsigned int w = 8; unsigned int x = 1; unsigned int y = 4; unsigned int z = 8]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:90:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct alignof error: NoteExpressionValueEvent
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:50:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (alignof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:90:1: note: in expansion of macro 'SMTG_TYPE_ALIGN_CHECK'
 SMTG_TYPE_ALIGN_CHECK (NoteExpressionValueEvent, 8, 1, 4, 8)
 ^~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::alignofCheckNoteExpressionTypeInfo<NoteExpressionTypeInfo, w, x, y, z>::alignofCheckNoteExpressionTypeInfo() [with NoteExpressionTypeInfo = Steinberg::Vst::NoteExpressionTypeInfo; unsigned int w = 8; unsigned int x = 1; unsigned int y = 8; unsigned int z = 8]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:92:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct alignof error: NoteExpressionTypeInfo
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:50:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (alignof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:92:1: note: in expansion of macro 'SMTG_TYPE_ALIGN_CHECK'
 SMTG_TYPE_ALIGN_CHECK (NoteExpressionTypeInfo, 8, 1, 8, 8)
 ^~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::sizeofCheckProcessContext<ProcessContext, w, x, y, z>::sizeofCheckProcessContext() [with ProcessContext = Steinberg::Vst::ProcessContext; unsigned int w = 112; unsigned int x = 104; unsigned int y = 112; unsigned int z = 112]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:97:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct sizeof error: ProcessContext
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:44:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (sizeof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:97:1: note: in expansion of macro 'SMTG_TYPE_SIZE_CHECK'
 SMTG_TYPE_SIZE_CHECK (ProcessContext, 112, 104, 112, 112)
 ^~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h: In instantiation of 'constexpr Steinberg::Vst::{anonymous}::alignofCheckProcessContext<ProcessContext, w, x, y, z>::alignofCheckProcessContext() [with ProcessContext = Steinberg::Vst::ProcessContext; unsigned int w = 8; unsigned int x = 1; unsigned int y = 8; unsigned int z = 8]':
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:100:1:   required from here
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:30:35: error: static assertion failed: Struct alignof error: ProcessContext
    static_assert (Operator (Type) ==                                                      \
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/pluginterfaces/base/typesizecheck.h:50:2: note: in expansion of macro 'SMTG_TYPE_STATIC_CHECK'
  SMTG_TYPE_STATIC_CHECK (alignof, Type, Platform64Size, MacOS32Size, Win32Size, Linux32Size)
  ^~~~~~~~~~~~~~~~~~~~~~
/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/public.sdk/source/vst/testsuite/vststructsizecheck.h:100:1: note: in expansion of macro 'SMTG_TYPE_ALIGN_CHECK'
 SMTG_TYPE_ALIGN_CHECK (ProcessContext, 8, 1, 8, 8)
 ^~~~~~~~~~~~~~~~~~~~~
make[4]: Leaving directory '/build/uhhyou-plugins-vst3-0.0.0+20200702/vst3sdk/build'
[ 59%] Built target IterativeSinCluster_dspcore_avx512
make[4]: *** [public.sdk/samples/vst-hosting/validator/CMakeFiles/validator.dir/build.make:635: public.sdk/samples/vst-hosting/validator/CMakeFiles/validator.dir/source/validator.cpp.o] Error 1
make[4]: *** Waiting for unfinished jobs....
```

_Note: it works with a 64 build._

**Comment by ryukau** [comment]
From the log, it seems like vst3sdk is failing to build. Could you build vst3sdk with following commands?

```bash
cmake -DCMAKE_BUILD_TYPE=Release ..
cmake --build .
```

Also try this one from [official vst3sdk repo](https://github.com/steinbergmedia/vst3sdk#build-the-examples-on-linux). However, I think following command builds with debug config.

```bash
cmake ..
cmake --build . --config Release
```

**Comment by ryukau** [comment]
Sorry, I forgot that VST 3 versions do not support SSE2. So they will not run on i686.

I omitted SSE2 because of macOS build fail and my inexperience of C++ macro at that time.

I might add SSE2 support in future but it's low priority. If you are interested in, take a look at following codes.

- `<PluginName>/CMakeLists.txt`
- `PlugProcessor::PlugProcessor()` in `<PluginName>/source/plugprocessor.cpp`
- `<PluginName>/source/dsp/dspcore.*`

In `dspcore.hpp` there are lines like following:

```c++
DSPCORE_CLASS(AVX512)
DSPCORE_CLASS(AVX2)
DSPCORE_CLASS(AVX)
```

In `dspcore.cpp` there are lines like following:

```c++
#if INSTRSET >= 10
#define DSPCORE_NAME DSPCore_AVX512
#elif INSTRSET >= 8
#define DSPCORE_NAME DSPCore_AVX2
#elif INSTRSET >= 7
#define DSPCORE_NAME DSPCore_AVX
#else
#error Unsupported instruction set
#endif
```

It needs to add conditions around these codes.

The `INSTRSET` macro definitioin comes from [vector class library](https://github.com/vectorclass/version2). See VCL version 2 manual section 9.9 "Instruction sets and CPU dispatching" for more details.

**Comment by ryukau** [comment]
Most of the plugins are not explicitly using SIMD anymore. So they might be able to build for 32 bit Linux.

One change is required for top level `CMakeLists.txt`. Following condition is only considering macOS, so adding detection of 32 bit Linux environment is necessary.

```cmake
if(NOT APPLE) # Change this.
  # These plugins are dependent on x86_64 specific SIMD.
  # ...
endif()
```

I will not actively fix this issue and add 32 bit support, because I don't use 32 bit environment. Also I'm not sure that the plugins works without performance problems. That said, if someone is interested into testing and patching, pull request is welcome.

**Comment by ryukau** [comment]
Closing because I won't spend more time on this issue.
